src/api/goods.ts

```
@@ -4,13 +4,37 @@ import { Good } from '../types/Good';
 const API_URL = `https://mate-academy.github.io/react_dynamic-list-of-goods/goods.json`;
 
 export function getAll(): Promise<Good[]> {
-  return fetch(API_URL).then(response => response.json());
+  return fetch(API_URL)
+    .then(response => {
+      if (response.ok) {
+        return response.json();
+      }
+
+      return 'A API está inacessível via protocolo HTTP.';
```

If the response is not ok, you return a string instead of rejecting the promise or throwing an error. This breaks the expected contract of returning a Promise<Good[]> and makes error handling in the UI unreliable. You should throw an error or use Promise.reject here. This violates checklist item #3: "Error handling must be implemented for server requests (e.g., handle network or response errors and reflect them appropriately in the app logic/UI)."